ios/MmkvHostObject.mm

```diff
@@ -10,10 +10,16 @@
 #import "MmkvHostObject.h"
 #import "JSIUtils.h"
 
-MmkvHostObject::MmkvHostObject(NSString* instanceId, NSString* path, NSString* cryptKey)
+MmkvHostObject::MmkvHostObject(NSString* instanceId, NSString* path, NSString* cryptKey, NSString* appGroup)
 {
+  NSString *groupDir = [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:appGroup].path;
   NSData* cryptData = cryptKey == nil ? nil : [cryptKey dataUsingEncoding:NSUTF8StringEncoding];
-  instance = [MMKV mmkvWithID:instanceId cryptKey:cryptData rootPath:path];
+  if (appGroup == nil) {
+    instance = [MMKV mmkvWithID:instanceId cryptKey:cryptData rootPath:path];
+  } else {
+    [MMKV initializeMMKV:nil groupDir:groupDir logLevel:MMKVLogNone];
```

Comment:
Hmm, this causes MMKV to re-initialize. Is that a safe thing to do? Right now we call `initializeMMKV` at the launch of the app... Maybe that's when the appGroup needs to be specified, instead of everytime you do new MMKV()? 🤔

Reply:
ah, I saw what you mean. I'll try to move and test it again. For now, it's working, but I don't know the side effect.

Reply:
@mrousavy btw if thats the cases, it seems we cannot put appGroup as a parameters in JS side. I'll try to detect in Native side instead of JS.